code/game/machinery/Freezer.dm

```diff
@@ -104,59 +104,46 @@
 		to_chat(user, "<span class='notice'>Close the maintenance panel first.</span>")
 		return
 
-	src.ui_interact(user)
+	tgui_interact(user)
 
-/obj/machinery/atmospherics/unary/cold_sink/freezer/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	// update the ui if it exists, returns null if no ui is passed/found
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/atmospherics/unary/cold_sink/freezer/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 	if(!ui)
-		// the ui does not exist, so we'll create a new() one
-        // for a list of parameters and their descriptions see the code docs in \code\modules\nano\nanoui.dm
-		ui = new(user, src, ui_key, "freezer.tmpl", "Gas Cooling System", 540, 300)
-		// open the new ui window
+		ui = new(user, src, ui_key, "ThermoMachine", "Gas Cooling System", 540, 200)
```

Comment:
Call it something other than `ThermoMachine`, because thats not descriptive enough. Maybe `GasFreezer`, since everything falls under the `freezer` path.

Reply:
I ask you to reconsider. The heater also uses this template, and I find calling that a gasfreezer is misleading. The circuit board also calls the freezer/heater a thermomachine, so the template name is consistent.

Reply:
The old nano template was `freezer.tmpl`. All of the code is inside `Freezer.dm`. Please make it `GasFreezer`.

Reply:
done